graders/c/cgrader/cgrader.py

```diff
@@ -452,17 +452,21 @@ def compile_re(t):
                 if must_match_all_outputs
                 else " one of"
             )
-            msg = f"Expected{comment}:\n\t" + "\n\t".join(
-                f"\033[32m{t}\033[0m"
-                if highlight_matches and r.search(outcmp) is not None
-                else t
+            msg = f"Expected{comment}:\n-\t" + "\n-\t".join(
+                str(
+                    f"\033[32m{t}\033[0m"
+                    if highlight_matches and r.search(outcmp) is not None
+                    else t
+                ).replace("\n", "\n\t")
```

Comment:
Help me understand what's actually happening here? What is the leading dash and why are tabs being used - seems like that would make the unexpected whitespace problem worse?

Reply:
Updated the original PR comment for clarity.

Reply:
Thanks for the extra description. I worry that the dashes are ambiguous, especially when only one output is present. When I was first looking at the example output and trying to parse the code, I thought perhaps that `-` was a diff-style line prefix indicating that the output was missing a line or something.

Looking specifically at the original linked issue, I worry that with this change, a student is going to think that either a) their output must start with a `-` or b) their output must have a leading tab on each line, or both.

Perhaps for the case of a single option, we don't show any item marker and don't include any tabs. It's less clear to me what to do with multiple options.

Reply:
Switched to blank line (`\n\n`) when any expected input has a line break.

Reply:
Discussed with Jonatan offline. If instructors don't like the default message, they can provide their own with `msg="..."` in their call to `test_run`.